server/base/src/test/java/org/apache/accumulo/server/conf/codec/VersionedPropEncryptCodecTest.java

```diff
@@ -0,0 +1,251 @@
+      dos.writeUTF("A");
+      dos.writeUTF("B");
+      dos.writeUTF("C");
+
+      cos.flush();
+      cos.close();
+
+      payload = bos.toByteArray();
+
+      log.debug("Output: {}", payload);
+
+    }
+
+    cipher.init(Cipher.DECRYPT_MODE, cipherProps.getSecretKey(), cipherProps.getParameterSpec());
+    try (ByteArrayInputStream bis = new ByteArrayInputStream(payload)) {
+
+      // write the property map keys, values.
+      try (CipherInputStream cis = new CipherInputStream(bis, cipher);
+
+          DataInputStream cdis = new DataInputStream(cis)) {
+
+        assertEquals("A", cdis.readUTF());
+        assertEquals("B", cdis.readUTF());
+        assertEquals("C", cdis.readUTF());
+      }
+    }
+  }
+
+  /**
+   * Validate versioning with something other than default.
+   */
+  @Test
+  public void roundTripEncryption() throws Exception {
+
+    int aVersion = 13;
+    Instant now = Instant.now();
+
+    Map<String,String> p = new HashMap<>();
+    p.put("k1", "v1");
+
+    VersionedProperties vProps = new VersionedProperties(aVersion, now, p);
+
+    final char[] pass = {'a', 'b', 'c'};
+    final byte[] salt = {1, 2, 3};
+
+    VersionedPropCodec encoder = VersionedPropEncryptCodec.codec(false,
+        new VersionedPropEncryptCodec.GCMCipherParams(pass, salt));
+
+    byte[] encodedBytes = encoder.toBytes(vProps);
+
+    log.debug("Encoded: {}", encodedBytes);
+
+    VersionedProperties decodedProps = encoder.fromBytes(encodedBytes);
+
+    log.debug("Decoded: {}", decodedProps.print(true));
+
+    assertEquals(vProps.getProperties(), decodedProps.getProperties());
+
+    // validate that the expected node version matches original version.
+    assertEquals(aVersion, vProps.getDataVersion());
+
+    // validate encoded version incremented.
+    assertEquals(aVersion + 1, decodedProps.getDataVersion());
+
+    assertEquals("encoded version should be 1 up", aVersion + 1, decodedProps.getDataVersion());
+    assertEquals("version written should be the source next version", vProps.getNextVersion(),
+        decodedProps.getDataVersion());
+    assertEquals("the next version in decoded should be +2", aVersion + 2,
+        decodedProps.getNextVersion());
+
+    assertTrue("timestamp should be now or earlier",
+        vProps.getTimestamp().compareTo(Instant.now()) <= 0);
+
+  }
+
+  /**
+   * Validate versioning with something other than default.
+   */
+  @Test
+  public void roundTripEncryptionCompressed() throws Exception {
+
+    int aVersion = 13;
+    Instant now = Instant.now();
+
+    // compression friendly
+    Map<String,String> p = new HashMap<>();
+    p.put("accumulo.prop.key_name.1", "value1");
+    p.put("accumulo.prop.key_name.2", "value2");
+    p.put("accumulo.prop.key_name.3", "value3");
+    p.put("accumulo.prop.key_name.4", "value4");
+    p.put("accumulo.prop.key_name.5", "value5");
+    p.put("accumulo.prop.key_name.6", "value9");
+
+    VersionedProperties vProps = new VersionedProperties(aVersion, now, p);
```

Comment:
```suggestion

    // compression friendly
    var p = Map.of("accumulo.prop.key_name.1", "value1", "accumulo.prop.key_name.2", "value2",
        "accumulo.prop.key_name.3", "value3", "accumulo.prop.key_name.4", "value4",
        "accumulo.prop.key_name.5", "value5", "accumulo.prop.key_name.6", "value9");

    VersionedProperties vProps = new VersionedProperties(aVersion, Instant.now(), p);
```

Reply:
`Map.of` in this case is a little less readable in my opinion.

Reply:
I'm going to leave this for now

Reply:
`Map.of` has the nice property of being immutable, so you don't have to worry that it's modified anywhere in the test. You can format it to get one pair per line and suppress formatting by wrapping it with `// @formatter:off` and `// @formatter:on`, but it's probably not worth it. I actually prefer two columns per line, as long as both key and value fit (and they do here) because I don't actually care about the data and want to see the subsequent lines of code sooner. But, either way is fine with me. It's definitely not important.